### ISSUE TITLE
Is flutter web support on the roadmap?

### ISSUE THREAD

**Issue by NoahHahm** [hydro-sdk/hydro-sdk#595]
desktop (macos, windows) builds successfully.
I have confirmed that there is no problem, but web support is not yet available, right?
the logs for both macos and windows environments are the same as below.
tested in two environments.

````
PS C:\Users\Noah\Documents\testapp> flutter run -d chrome 
Launching lib\main.dart on Chrome in debug mode...
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:223:68: Error: Superclass has no constructor named 'Path'.
  RTManagedPath({required this.table, required this.hydroState}) : super() {
                                                                   ^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:229:21: Error: Superclass has no getter named 'fillType'.
      return [super.fillType];
                    ^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:232:13: Error: Superclass has no setter named 'fillType'.
      super.fillType =
            ^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:237:13: Error: Superclass has no method named 'moveTo'.
      super.moveTo(args[1]?.toDouble(), args[2]?.toDouble());
            ^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:241:13: Error: Superclass has no method named 'relativeMoveTo'.
      super.relativeMoveTo(args[1]?.toDouble(), args[2]?.toDouble());
            ^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:245:13: Error: Superclass has no method named 'lineTo'.
      super.lineTo(args[1]?.toDouble(), args[2]?.toDouble());
            ^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:249:13: Error: Superclass has no method named 'relativeLineTo'.
      super.relativeLineTo(args[1]?.toDouble(), args[2]?.toDouble());
            ^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:254:13: Error: Superclass has no method named 'quadraticBezierTo'.
      super.quadraticBezierTo(args[1]?.toDouble(), args[2]?.toDouble(), 
            ^^^^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:260:13: Error: Superclass has no method named 'relativeQuadraticBezierTo'.
      super.relativeQuadraticBezierTo(args[1]?.toDouble(), args[2]?.toDouble(),
            ^^^^^^^^^^^^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:265:13: Error: Superclass has no method named 'cubicTo'.
      super.cubicTo(
            ^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:276:13: Error: Superclass has no method named 'relativeCubicTo'.
      super.relativeCubicTo(
            ^^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:286:13: Error: Superclass has no method named 'conicTo'.
      super.conicTo(args[1]?.toDouble(), args[2]?.toDouble(),
            ^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:292:13: Error: Superclass has no method named 'relativeConicTo'.
      super.relativeConicTo(args[1]?.toDouble(), args[2]?.toDouble(),   
            ^^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:297:13: Error: Superclass has no method named 'arcTo'.
      super.arcTo(
            ^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:305:13: Error: Superclass has no method named 'arcToPoint'.
      super.arcToPoint(
            ^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:316:13: Error: Superclass has no method named 'relativeArcToPoint'.
      super.relativeArcToPoint(
            ^^^^^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:326:13: Error: Superclass has no method named 'addRect'.
      super.addRect(
            ^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:331:13: Error: Superclass has no method named 'addOval'.
      super.addOval(
            ^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:336:13: Error: Superclass has no method named 'addArc'.
      super.addArc(
            ^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:343:13: Error: Superclass has no method named 'addPolygon'.
      super.addPolygon(
            ^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:350:13: Error: Superclass has no method named 'addRRect'.
      super.addRRect(
            ^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:355:13: Error: Superclass has no method named 'addPath'.
      super.addPath(
            ^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:363:13: Error: Superclass has no method named 'extendWithPath'.
      super.extendWithPath(
            ^^^^^^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:371:13: Error: Superclass has no method named 'close'.
      super.close();
            ^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:375:13: Error: Superclass has no method named 'reset'.
      super.reset();
            ^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:380:15: Error: Superclass has no method named 'contains'.
        super.contains(maybeUnBoxAndBuildArgument<Offset>(args[1],
              ^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:387:27: Error: Superclass has no method named 'shift'.
            object: super.shift(maybeUnBoxAndBuildArgument<Offset>(args[1],
                          ^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:396:27: Error: Superclass has no method named 'transform'.
            object: super.transform(maybeUnBoxAndBuildArgument<Float64List>(
                          ^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:406:27: Error: Superclass has no method named 'getBounds'.
            object: super.getBounds(),
                          ^^^^^^^^^
/C:/tools/flutter/.pub-cache/git/hydro-sdk-a248c30683c2e39f55a6916deb7e791ef557f9c5/lib/cfr/builtins/libs/dart/ui/path.dart:414:27: Error: Superclass has no method named 'computeMetrics'.
            object: super.computeMetrics(forceClosed: args[1]['forceClosed']),
                          ^^^^^^^^^^^^^^
Unhandled exception:                                                    
NoSuchMethodError: The getter 'name' was called on null.
Receiver: null
Tried calling: name
#0      Object.noSuchMethod (dart:core-patch/object_patch.dart:54:5)
#1      ProgramCompiler._emitSuperConstructorCallIfNeeded (package:dev_compiler/src/kernel/compiler.dart:1598:14)  
#2      ProgramCompiler._emitConstructorBody (package:dev_compiler/src/kernel/compiler.dart:1552:19)
#3      ProgramCompiler._emitConstructor.<anonymous closure>.<anonymous closure>
(package:dev_compiler/src/kernel/compiler.dart:1508:19)
#4      ProgramCompiler._superDisallowed (package:dev_compiler/src/kernel/compiler.dart:3459:24)
#5      ProgramCompiler._emitConstructor.<anonymous closure>
(package:dev_compiler/src/kernel/compiler.dart:1507:15)
#6      ProgramCompiler._withLetScope (package:dev_compiler/src/kernel/compiler.dart:2266:25)
#7      ProgramCompiler._withCurrentFunction (package:dev_compiler/src/kernel/compiler.dart:3449:18)
#8      ProgramCompiler._emitConstructor (package:dev_compiler/src/kernel/compiler.dart:1505:16)
#9      ProgramCompiler._defineConstructors (package:dev_compiler/src/kernel/compiler.dart:1147:38)
#10     ProgramCompiler._emitClassDeclaration (package:dev_compiler/src/kernel/compiler.dart:669:19)
#11     ProgramCompiler._emitClass (package:dev_compiler/src/kernel/compiler.dart:592:21)
#12     List.forEach (dart:core-patch/growable_array.dart:403:8)
#13     ProgramCompiler._emitLibrary (package:dev_compiler/src/kernel/compiler.dart:541:23)
#14     List.forEach (dart:core-patch/growable_array.dart:403:8)
#15     ProgramCompiler.emitModule (package:dev_compiler/src/kernel/compiler.dart:401:15)
#16     JavaScriptBundler.compile (package:frontend_server/src/javascript_bundle.dart:146:33)
<asynchronous suspension>
#17     FrontendCompiler.writeJavascriptBundle (package:frontend_server/frontend_server.dart:638:32)
<asynchronous suspension>
#18     FrontendCompiler.compile (package:frontend_server/frontend_server.dart:549:9)
<asynchronous suspension>
#19     listenAndCompile.<anonymous closure> (package:frontend_server/frontend_server.dart:1119:11)
<asynchronous suspension>
the Dart compiler exited unexpectedly.                                  
Waiting for connection from debug service on Chrome...             15.8s
Failed to compile application.
````

**Comment by chgibb** [comment]
What version of Flutter are you using?

**Comment by NoahHahm** [comment]
2.2.1 I used it and I think it will be reproduced in the latest stable version

**Comment by chgibb** [comment]
The class `RTManagedPath` in `lib/cfr/builtins/libs/dart/ui/path.dart` provides a binding exposing the `Path` class from `dart:ui` to code running in CFR. It does this by subclassing `Path`.

`Path` is defined for all targets (except web) here: https://github.com/flutter/engine/blob/master/lib/ui/painting.dart#L2280
And for web here: https://github.com/flutter/engine/blob/master/lib/web_ui/lib/src/ui/path.dart#L10

The Flutter web implementation of `Path` is vastly different than its counterpart for other targets. I've opened an issue describing this here: https://github.com/flutter/flutter/issues/87861

I want to wait and see if this is something that can be changed on Flutter's end. If not, this can be fixed on Hydro-SDKs end as part of the work done for this feature https://github.com/hydro-sdk/hydro-sdk/issues/563 and the `dart:ui` project https://github.com/hydro-sdk/hydro-sdk/projects/17

**Comment by chgibb** [comment]
There doesn't appear to be very much movement on this from the Flutter team's end. The features required to fix this from within Hydro-SDK have landed as part of 0.0.1-nightly.315 in PR #609 . @NoahHahm is Flutter web a requirement for you?

**Comment by NoahHahm** [comment]
It's not something you need urgently. I was just trying to make a demo with a cross-platform introduction in the internal introduction.
thank you
